src/dune_rules/dep_rules.ml

```diff
@@ -10,13 +10,13 @@ let ooi_deps { vimpl; sctx; dir; obj_dir; modules = _; stdlib = _; sandbox = _ }
     ~dune_version ~vlib_obj_map ~(ml_kind : Ml_kind.t) (m : Module.t) =
   let cm_kind =
     match ml_kind with
-    | Intf -> Cm_kind.Cmi
+    | Intf -> (* todo : fix *) Lib_mode.Cm_kind.Melange Cmi
     | Impl -> vimpl |> Option.value_exn |> Vimpl.impl_cm_kind
   in
   let write, read =
     let ctx = Super_context.context sctx in
     let unit =
-      Obj_dir.Module.cm_file_exn obj_dir m ~kind:(Ocaml cm_kind) |> Path.build
+      Obj_dir.Module.cm_file_exn obj_dir m ~kind:cm_kind |> Path.build
```

Comment:
Note that the ocamlobjinfo path is only needed for implementing a virtual library that is installed. We don't yet have the installation of melange libraries, so we won't be able to fix this yet.

Reply:
Thanks! I started tweaking code from here, then fixed the objs part, but I wasn't sure this needed changing.